include/xtensor/xmath.hpp

```diff
@@ -407,6 +407,45 @@ namespace detail {
                           init_value_fct(detail::fill_init<init_value_type >(INIT))), std::forward<E>(e), axes, es); \
     }
 
+#define XTENSOR_MIN_MAX_REDUCER_FUNCTION(NAME, FUNCTOR, INIT_VALUE_TYPE, INIT)                                       \
```

Comment:
We should disable the return type for `nanmin` and `nanmax` because the following code will have undefined behavior (https://stackoverflow.com/questions/10366485/problems-casting-nan-floats-to-int):
```
xarray<double> all_nan = {{ nanv, nanv  }, { nanv, nanv }};
nanmax<int>(all_nan)())
```
In the `numpy` version [nanmin](https://numpy.org/doc/stable/reference/generated/numpy.nanmin.html), unlike other nan functions like `nansum`, it is also not allowed to specify the return type `dtype`.

The new macro could be possibly applied to `amin` and `amax` as it does not make too much sense for the two to have a return type template.

Reply:
I think it's fine to keep the return type actually. Even if it's undefined behavior, C++ does not *forbid* you to cast `nan` values to integers and I would be in favor of sticking with this behavior. Besides, there might be use cases where the uer wants to cast from a floating point type to another, and not having the return type would prevent this behavior. A warining can be added in the documentation so that the user is aware of the possible undefined behavior when casting to int.